core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -140,7 +140,7 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
       // SPARK-28843: limit the OpenMP thread pool to the number of cores assigned to this executor
       // this avoids high memory consumption with pandas/numpy because of a large OpenMP thread pool
       // see https://github.com/numpy/numpy/issues/10455
-      execCoresProp.foreach(envVars.put("OMP_NUM_THREADS", _))
+      envVars.put("OMP_NUM_THREADS", conf.get("spark.task.cpus", "1"))
```

Comment:
Okay, maybe I was overthinking. I think this fix makes sense. I thought that we might need to introduce new property to the resource profiler (like `resource.executor.cores`).

Reply:
I think we should remove the code here,
because we already set this in here https://github.com/apache/spark/blob/74410ca2f1318177e558f1e719e0cac0f0196807/core/src/main/scala/org/apache/spark/SparkContext.scala#L551

Reply:
This change is logical. The only argument for the original setting is it might actually help some tasks to fully use the CPUs by over-committing (assuming not all tasks are using all these threads at the same time). However it hurts throughput when closer to saturated.

Reply:
It will. One OMP call will not leverage all cores available for this executor, as its thread pool is set to 1 by default for `spark.task.cpus=1` but `spark.executor.cores=10`.

Reply:
Right, I'm saying that sometimes 1 is actually worse, but probably better as a default, yes

Reply:
@WeichenXu123 @HyukjinKwon During SPARK-41188, have you considered setting default to the number of executor cores? Have we seen the number of executor cores set to very high in production (especially ML)?

Reply:
There are 2 choices:

- Merge this PR and add a release note to state this subtle change
- Reject this PR and modify SPARK-41188 change in SparkContext (for Standalone cluster manager) to set default 
OMP_NUM_THREADS to executor cores

It is impossible for the default value to satisfy all use cases. With either choice, some users will have to set OMP_NUM_THREADS explicitly to balance performance and python worker memory pressure.

Reply:
I am good with the current proposal but maybe worthwhile adding it to a release note.

FWIW, for the very initial PR, I was worried about limiting the number of cores (and misunderstood this concept a bit there) and its potential impact. After having multiple syncs with ML guys, now I understand it better :-).

Reply:
Thanks for the context!
Good to go for me.